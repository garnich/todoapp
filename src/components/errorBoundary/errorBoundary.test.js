import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ErrorBoundary from './errorBoundary'

Enzyme.configure({ adapter: new Adapter() })

const children = <h1>test</h1>
describe('Test ErrorBoundary WITHOUT errors', () => {
  it('<ErrorBoundary> test snapshot', () => {
    const wrapper = shallow(<ErrorBoundary>{children}</ErrorBoundary>)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('Test ErrorBoundary WITH errors', () => {
  const wrapper = shallow(<ErrorBoundary>{children}</ErrorBoundary>)

  beforeEach(() => {
    wrapper.instance().componentDidCatch('error')
    wrapper.update()
  })

  it('<ErrorBoundary> test snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update the state to indicate an error', () => {
    expect(wrapper.instance().state.hasError).toBeTruthy()
  })

  it('should not render the child component', () => {
    expect(wrapper.find('h1').exists()).toBeFalsy()
  })
})
