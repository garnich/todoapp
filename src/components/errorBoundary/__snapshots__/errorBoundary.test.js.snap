// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ErrorBoundary WITH errors <ErrorBoundary> test snapshot 1`] = `
<p
  className="alert alert-danger"
/>
`;

exports[`Test ErrorBoundary WITHOUT errors <ErrorBoundary> test snapshot 1`] = `
<h1>
  test
</h1>
`;
